packages/typescript-estree/src/parseSettings/resolveProjectList.ts

```diff
@@ -0,0 +1,142 @@
+import debug from 'debug';
+import { sync as globSync } from 'globby';
+import isGlob from 'is-glob';
+
+import type { CanonicalPath } from '../create-program/shared';
+import {
+  createHash,
+  ensureAbsolutePath,
+  getCanonicalFileName,
+} from '../create-program/shared';
+import type { TSESTreeOptions } from '../parser-options';
+import {
+  DEFAULT_TSCONFIG_CACHE_DURATION_SECONDS,
+  ExpiringCache,
+} from './ExpiringCache';
+
+const log = debug(
+  'typescript-eslint:typescript-estree:parser:parseSettings:resovleProjectGlobs',
+);
+
+let RESOLUTION_CACHE: ExpiringCache<string, readonly CanonicalPath[]> | null =
+  null;
+
+/**
+ * Normalizes, sanitizes, resolves and filters the provided project paths
+ */
+export function resolveProjectList(
+  options: Readonly<{
+    cacheLifetime?: TSESTreeOptions['cacheLifetime'];
+    project: TSESTreeOptions['project'];
+    projectFolderIgnoreList: TSESTreeOptions['projectFolderIgnoreList'];
+    singleRun: boolean;
+    tsconfigRootDir: string;
+  }>,
+): readonly CanonicalPath[] {
+  const sanitizedProjects: string[] = [];
+
+  // Normalize and sanitize the project paths
+  if (typeof options.project === 'string') {
+    sanitizedProjects.push(options.project);
+  } else if (Array.isArray(options.project)) {
```

Comment:
Maybe a good place to add a `TODO(#....): log a warning if a project is not a string`, and file an issue?